pipeline/matching/generators/limited_schedules.py

```diff
@@ -1,19 +1,56 @@
-from typing import Iterator
+from typing import Dict, Iterator, List, Tuple
 
 from pipeline.matching.core import MatchGenerator
-from pipeline.types import Match, MatchingInput
+from pipeline.types import Match, MatchingInput, MatchMetadata
 
 GENERATOR_LIMITED_SCHEDULES = "limitedSchedulesGenerator"
 
 
 class LimitedSchedulesGenerator(MatchGenerator):
-    def __init__(self, max_available: int = 0):
+    def __init__(self, max_available: int = 1):
         # Maximum number of available slots in a user's schedule for them to be
         # eligible for a limited availability match.
         self.max_available = max_available
         super().__init__(name=GENERATOR_LIMITED_SCHEDULES)
 
+    def create_matches(self, users_availability: Dict) -> List[Match]:
+        matched_users = []
+        for userid_tuple, avail in users_availability.items():
+            metadata = MatchMetadata(
+                generator=GENERATOR_LIMITED_SCHEDULES, availability=avail
+            )
+            users = set(userid_tuple)
+            match = Match(users=users, metadata=metadata)
+            matched_users.append(match)
+        return matched_users
+
     def generate(self, inp: MatchingInput) -> Iterator[Match]:
-        # TODO: Implement your generator
-        inp.logger.info("Luke Arrington was here!")
-        yield from []
+        # get limited availability users
+        limited_availability = []
+        for user in inp.users:
+            if len(user.schedule) <= self.max_available:
+                limited_availability.append(user)
+
+        users_availability: Dict[Tuple, List] = {}
+        # compare limited users to all users
+        for limited_user in limited_availability:
+            for user in inp.users:
+                userid_tuple = (limited_user.uid, user.uid)
```

Comment:
There's also a subtle issue here that if both users are limited users, we might create two versions of the same match, one where the first user comes first in the tuple, and one with the opposite order.

This is okay because the matching engine ignores duplicate matches. But we can fix this in the future when you write more test cases.